utils/mkhtml.py

```diff
@@ -188,71 +179,169 @@ def download_git_commit(url, response_format, *args, **kwargs):
-            git_log["date"] = time.ctime(os.path.getmtime(src_dir))
-        stdout, stderr = subprocess.Popen(
-            args=["git", "log", "-1", src_dir],
-            stdout=subprocess.PIPE,
+
+def format_git_commit_date_from_rest_api(
+    commit_datetime, datetime_format="%A %b %d %H:%M:%S %Y"
+):
+    """Format datetime from remote GitHub REST API
+
+    :param str commit_datetime: commit datetime
+    :param str datetime_format: output commit datetime format
+                                e.g. Sun Jan 16 23:09:35 2022
+
+    :return str: output formatted commit datetime
+    """
+    return datetime.strptime(
+        commit_datetime,
+        "%Y-%m-%dT%H:%M:%SZ",  # ISO 8601 YYYY-MM-DDTHH:MM:SSZ
+    ).strftime(datetime_format)
+
+
+def format_git_commit_date_from_local_git(
+    commit_datetime, datetime_format="%A %b %d %H:%M:%S %Y"
+):
+    """Format datetime from local Git or JSON file
+
+    :param str commit_datetime: commit datetime
+    :param str datetime_format: output commit datetime format
+                                e.g. Sun Jan 16 23:09:35 2022
+
+    :return str: output formatted commit datetime
+    """
+    return datetime.fromtimestamp(
+        int(commit_datetime),
+    ).strftime(datetime_format)
+
+
+def has_src_code_git(src_dir):
+    """Has core module or addon source code Git
+
+    :param str src_dir: core module or addon root directory
+
+    :return subprocess.CompletedProcess or None: subprocess.CompletedProcess
+                                                 if core module or addon
+                                                 source code has Git
+    """
+    try:
+        process_result = subprocess.run(
+            ["git", "log", "-1", "--format=%H,%at", src_dir],
             stderr=subprocess.PIPE,
-        ).communicate()
-        stdout = decode(stdout)
-        stderr = decode(stderr)
-
-        if stderr and "fatal: not a git repository" in stderr:
-            response = download_git_commit(
-                url=grass_addons_url if is_addon else grass_modules_url,
-                response_format="application/json",
+            stdout=subprocess.PIPE,
+        )  # --format=%H,%at commit hash,author date (UNIX timestamp)
+        return process_result if process_result.returncode == 0 else None
+    except FileNotFoundError:
+        return None
+
+
+def get_last_git_commit(src_dir, addon_path):
+    """Get last module/addon git commit
+
+    :param str src_dir: module/addon source dir
+    :param str addon_path: addon path
+
+    :return dict git_log: dict with key commit and date, if not
+                          possible download commit from GitHub REST API
+                          server values of keys have "unknown" string
+    """
+    process_result = has_src_code_git(src_dir=src_dir)
+    if process_result:
+        stdout = process_result.stdout.decode()
+        if stdout:
+            return parse_git_commit(commit=stdout)
+    else:
```

Comment:
I'm still not sure if this else should be here. What happens when stdout above is empty? If you just not test it, that may be fine given the context, but this seems like missing return (or raise).